lib/src/rules/curly_braces_in_flow_control_structures.dart

```diff
@@ -86,17 +86,22 @@ class _Visitor extends SimpleAstVisitor {
 
   @override
   void visitIfStatement(IfStatement node) {
-    if (node.elseStatement == null) {
+    var elseStatement = node.elseStatement;
+    if (elseStatement == null) {
       if (node.thenStatement is Block) return;
 
       final unit = node.root as CompilationUnit;
       if (unit.lineInfo.getLocation(node.rightParenthesis.end).lineNumber !=
           unit.lineInfo.getLocation(node.thenStatement.end).lineNumber) {
         rule.reportLint(node.thenStatement);
       }
-    } else if (node.elseStatement is! IfStatement) {
+    } else {
       _check(node.thenStatement);
-      _check(node.elseStatement);
+      if (elseStatement is IfStatement) {
+        _check(elseStatement.thenStatement);
```

Comment:
If `elseStatement` is an if statement, then I believe we will end up checking the then statement twice: once here and once when `visitIfStatement` is invoked with `elseStatement` as the argument. I think this just wants to be
```
if (elseStatement is! IfStatement) {
  _check(elseStatement);
}
```

Reply:
yes the elseStatement is checked twice but it's not the same check. Actually the `if (elseStatement == null)` should also check if `node.parent` is a IfStatement to avoid this case.

Reply:
It wasn't the else statement I was concerned about. Consider the following code:
```dart
if (condA)
  stmtA;
else if (condB)
  stmtB;
else
  stmtC;
```
When this visit method is visiting the if whose condition is `condA` the else statement will not be null, so it will invoke `_check` on `stmtA` (line 101). Because the else statement is another if statement (the one whose condition is `condB`), it will then check that if statement's then statement, which is `stmtB`.

Later, when this visit method is visiting the if statement whose condition is `condB`, it will invoke `_check` on `stmtB` (line 101).

Reply:
You're right. Thanks. Updated.